### PR TITLE
Use d2l-dialog-fullscreen component instead of overlay

### PR DESCRIPTION
**Pull request by annabelsegalld2l**
Updating the show all courses/course banner widget to use the d2l-dialog-fullscreen component rather than d2l-simple-overlay.

This is part of a set of PRs all needed to make the image-selector component work with d2l-dialog-fullscreen: 

- https://github.com/Brightspace/d2l-my-courses-ui/pull/892
- https://github.com/Brightspace/d2l-image-banner-overlay/pull/112
- https://github.com/Brightspace/image-selector/pull/48

### CODE REVIEW

File: d2l-basic-image-selector.js [hunk 1]
```diff
@@ -109,6 +109,10 @@ Polymer({
 	is: 'd2l-basic-image-selector',
 	ready: function() {
 		this._images = this._defaultImages;
+		this._initialize = this._initialize.bind(this);
+		this._clear = this._clear.bind(this);
+		window.addEventListener('d2l-dialog-open', this._initialize);
```
Comment (annabelsegalld2l) [inline d2l-basic-image-selector.js:114]: For now, the image selector will listen to the legacy overlay events and the new dialog events, so either one can work with the selector.

Comment (svanherk) [reply d2l-basic-image-selector.js:114]: Do these need to be set on the `window`?  It looks like before they were just set using the Polymer `listeners` array, which I _think_ is the same as using `this.addEventListener('d2l-dialog-open', this._initialize);`.  We usually want to set event listeners at the lowest level possible, and delay wire-up if possible so we don't block time-to-first-paint.

Comment (svanherk) [reply d2l-basic-image-selector.js:114]: For that to work, you might need to move the wire-up to `attached` (Polymer 1's wrapper of `connectedCallback` - after that they decided to just "use the platform".  You should be able to switch this from `attached` to `connectedCallback` and `detached` to `disconnectedCallback` with no problems).  Or you can try adding the new entries to the `listeners` array.  In all these cases, I think you would no longer need the binding calls.

Comment (svanherk) [reply d2l-basic-image-selector.js:114]: Darn, I think I see how this was working before... magic: https://github.com/Brightspace/simple-overlay/blob/master/d2l-simple-overlay.js#L145-L157

Comment (svanherk) [reply d2l-basic-image-selector.js:114]: Talking offline, we're leaning towards instead exposing the `initialize` and `clear` functions, so the consumer can decide when to call them (like, maybe, when opening and closing the dialog they put the image selector in).  @ryantmer Do you remember any gotchas/reasons why it wasn't done that way initially?

Comment (ryantmer) [reply d2l-basic-image-selector.js:114]: I'll be honest, I don't remember much of anything at all (like in general), but I really definitely don't remember this, sorry 😄 I've lost basically all Polymer knowledge at this point 😬 (Honestly I should probably not be a codeowner here haha, I'm pretty useless)

File: d2l-basic-image-selector.js [hunk 2]
```diff
@@ -226,7 +226,7 @@ Polymer({
 			this.organization = SirenParse(organization);
 		}
 	},
-	_initialize: function() {
+	initializeSearch: function() {
```
Comment (annabelsegalld2l) [inline d2l-basic-image-selector.js:229]: Now, the only change should be making these methods public. The mycourses and course banner widgets can then call the initialize and clear functions as needed.

**Review by svanherk — APPROVED** [review]
Looks like you might want to bump the `package.json` version in this PR, and then make a manual release after.

**Pull request merged**